### PR TITLE
Log N lookup for put and get key, value?

### PR DESCRIPTION
**Pull request by stienis96**
Would be nice to do this from the command line or something.
It seems like memcslap can't connect or something

### CODE REVIEW

**Comment by l8518** [comment]
@stienis96 Currently not woring.
Used this:
`sudo apt install libmemcached-tools`
with `memcslap --servers=localhost:11211 --execute-number=1 --test=get --concurrency=1`

**Comment by l8518** [comment]
@stienis96  Maybe this one helps on macOs https://formulae.brew.sh/formula/libmemcached

**Comment by stienis96** [comment]
I got it running on ubuntu now.

First building the project.
`mvn clean package -DskipTests`

After that running one central node with: `java -Dconfig.resource=/centralNode.conf -jar chord-1.0-allinone.jar`

And then memcslap: `memcslap --servers=localhost:11211 --execute-number=1 --test=get --concurrency=1`

**Comment by l8518** [comment]
just merged latest develop, as I'll branch of here

**Comment by l8518** [comment]
Also, i checked: https://github.com/RedisLabs/memtier_benchmark/tree/1.2.17

works with  `memtier_benchmark --protocol=memcache_text --port 11211`

**Comment by stienis96** [comment]
> Also, i checked: https://github.com/RedisLabs/memtier_benchmark/tree/1.2.17
> 
> works with `memtier_benchmark --protocol=memcache_text --port 11211`

Nice! So it is ready to merge?

**Comment by l8518** [comment]
> > Also, i checked: https://github.com/RedisLabs/memtier_benchmark/tree/1.2.17
> > works with `memtier_benchmark --protocol=memcache_text --port 11211`
> 
> Nice! So it is ready to merge?

Don't think so, as we do not fulfil the proctol just yet. starting on this now

**Comment by l8518** [comment]
@stienis96 FYI:
https://stackoverflow.com/questions/35219535/using-memtier-benchmark-every-key-is-missed

Not sure if we need that, currently we have 100% misses, but I guess that's alright?!

**Comment by stienis96** [comment]
@l8518 That's probably because the generated key is way bigger than the keys in the chord network right?

**Comment by stienis96** [comment]
image
I'm quite sure this is incorrect though..

**Comment by l8518** [comment]
@stienis96 Now it's normal i guess, however, the system does not distributed keys at all (all put on a single node). You did implement that key lookup, any ideas why it not works?

**Comment by l8518** [comment]
I guess, the hash key (for such small m) does not distribute very well?!

File: src/main/java/org/distributed/systems/chord/actors/NodeActor.java [hunk 1]
```diff
@@ -188,26 +229,26 @@ public Receive createReceive() {
                     }
                 })
                 .match(Connected.class, conn -> {
-                    System.out.println("MemCache Client connected");
                     manager.tell(conn, getSelf());
                     ActorRef memcacheHandler = getContext().actorOf(Props.create(MemcachedActor.class, storageActorRef));
```
Comment (l8518) [inline src/main/java/org/distributed/systems/chord/actors/NodeActor.java]: Is this the issue? However, if I change this to `getSelf()` everything goes to the shit. :sweat_smile: 

I would assume it works like this MemCacheActor -> Node -> StorageActor.
Currently it goes like MemCacheActor -> StorageActor...

I am clueless though :(

Comment (l8518) [reply src/main/java/org/distributed/systems/chord/actors/NodeActor.java]: @stienis96 Any idea?

Comment (stienis96) [reply src/main/java/org/distributed/systems/chord/actors/NodeActor.java]: @l8518 Yes that was mostly the issue. I can't test it fully now though, my ubuntu vm is a bit limited. Can you run your test one more time with a few nodes?

Comment (l8518) [reply src/main/java/org/distributed/systems/chord/actors/NodeActor.java]: :fire: :dancing_men:

**Pull request merged**